app/services/metadata_enhancer.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+# Enhances a DataWorks schema metadata record based on the metadata on that record and related sources
+class MetadataEnhancer
+  def self.call(...)
+    new(...).call
+  end
+
+  def initialize(mapped_record:)
+    @mapped_record = mapped_record.with_indifferent_access
+  end
+
+  # @return enhanced metadata record for the dataset with additional metadata
+  def call
+    add_stanford_authors
+    @mapped_record
+  end
+
+  # Match stanford contributors and creators with Stanford authorrsbased on ORCIDs
+  def add_stanford_authors
+    # Do we have any ORCIDs?
+    contributors = (@mapped_record[:creators] || []).concat(@mapped_record[:contributors] || [])
```

Comment:
fwiw for these type of things I like to do `Array(@mapped_record[:creators])` which has the same behavior (if it's empty, just give me an empty array)